src/socket.rs

```diff
@@ -382,3 +418,65 @@ impl From<mpsc::error::SendError<StateCommand>> for DisconnectError {
         DisconnectError::SocketShutdown
     }
 }
+
+#[derive(Clone)]
+struct ArcStatus {
+    status: Arc<AtomicU8>,
+    tx: broadcast::Sender<Result<Status, Arc<tungstenite::Error>>>,
+}
+impl ArcStatus {
+    fn new() -> Self {
+        let (tx, _) = broadcast::channel(5);
+
+        ArcStatus {
+            status: Arc::new(AtomicU8::new(Status::default() as u8)),
+            tx,
+        }
+    }
+
+    fn get(&self) -> Status {
+        unsafe { core::mem::transmute::<u8, Status>(self.status.load(Ordering::SeqCst)) }
+    }
+
+    fn set(&self, status: Status) {
+        let status_u8 = status as u8;
+        let previous_status_u8 = self.status.swap(status_u8, Ordering::SeqCst);
+
+        if status_u8 != previous_status_u8 {
```

Comment:
This can be `Ordering::AcqRel`

```rust
        if self.status.swap(status as u8, Ordering::AcqRel) != status as u8 {
            self.tx.send(Ok(status)).ok();
        }
```

Reply:
✅